src/leanback/java/com/archos/mediacenter/video/leanback/MainFragment.java

```diff
@@ -656,6 +661,28 @@ else if (cursorLoader.getId() == LOADER_ID_NON_SCRAPED_VIDEOS_COUNT) {
     @Override
     public void onLoaderReset(Loader<Cursor> cursorLoader) { }
 
+    private boolean isVideosListModified(Cursor oldCursor, Cursor newCursor) {
```

Comment:
I have a performance question: is it viable to check if there is a change in video collection each time we get to MainFragment if you have a very large video collection (e.g. 5k movies, 20k tv serie episodes) since basically you do a walk through the entire collection? If it confirms to be an issue, a suboptimal (good enough?) option would be to only make the decision on getCount basis.